### ISSUE TITLE
Auth and Pool Issues #2

### ISSUE THREAD

**Issue by BackedUpBooty** [standardnotes/standalone#7]
Hi there, the original "Auth and Pool Issues #1" was closed without resolving other than spinning up a new VM (not an option for me) so opening this. I have a similar issue. I'm using the standalone repo, running it on a synology NAS 920+ in docker. Followed the ./server.sh instructions after cloning the repo. Modified the .env with secrets and auth etc., changing `EXPOSED_PORT` to 3333 to avoid a conflict. I changed the `ENCRYPTION_SERVER_KEY` in auth.env to a 32 character random string. I've done `./server.sh update` and a `./server.sh cleanup` but as far as I can tell logs look the same and I can't resolve localhost:3333/healthcheck.

IP addresses for reference:

- 172.25.0.2 cache
- 172.25.0.3 api-gateway
- 172.25.0.4 auth-worker
- 172.25.0.5 auth
- 172.25.0.6 syncing-server-js-worker
- 172.25.0.7 db
- 172.25.0.8 syncing-server-js

**Logs** (using pastebin for larger logs):

from syncing-server-js
https://pastebin.com/DGZ8CMw5
TL:DR: pool cluster error, access denied for user@auth-worker IP; pool does Not exists [sic]

from syncing-server-js-worker
```nc: db (172.25.0.6:3306): Operation timed out,
nc: db (172.25.0.7:3306): Operation timed out,
db:3306 is unavailable yet - waiting for it to start
```

from db
https://pastebin.com/zvRZmcrd
TL:DR: [Note] mysqld: ready for connections.

from cache
https://pastebin.com/CMufHMP0
TL:DR: ready to accept connections

from auth
```
syncing-server-js:3000 is unavailable yet - waiting for it to start
```

from auth-worker AND api-gateway
```
auth:3000 is unavailable yet - waiting for it to start
```

Will sharing the .env file help?

**Comment by karolsojko** [comment]
@BackedUpBooty can you paste which versions of the services you have in your docker-compose file?

**Comment by BackedUpBooty** [comment]
Sure:

1. standardnotes/syncing-server-js:1.27.4 (server-js + server-js-worker)
2. standardnotes/api-gateway:1.6.0
3. standardnotes/auth:1.7.5 (auth + auth-worker)
4. mysql:5.6

**Comment by karolsojko** [comment]
> `ENCRYPTION_SERVER_KEY` in auth.env to a 32 character random string.

make sure this is a hex string. 

@BackedUpBooty can you try `./server.sh update` if the latest version fixes the issue? Also make sure to update your .env files since there were changes to them.

**Comment by BackedUpBooty** [comment]
Hey thanks for the reply. 

> make sure this is a hex string

I went to https://www.browserling.com/tools/random-hex, had it set to 32 digits and 5 options, ran the generator then selected one of the strings at random. Double checking that's correct?

Will running `./server.sh update` also update the .env files? I can't tell if you're saying that I will need to redo the changes I made to the .env files, or if the repo has made changes I'll need to review.

**Comment by karolsojko** [comment]
We're moving some of the authorisation and authentication functionalities from `syncing-server-js` to `auth` thus moved the env variables from `.env.sample` to `docker/auth.env.sample` 

The startup script will notify you that you need to update your differences, so you'll need to review the sample files and adjust accordingly.

**Comment by BackedUpBooty** [comment]
Got it, thanks for the info. Tbh as I don't actually have any data in the volume I may just start again with cloning your updated repo, after taking note of the few .env changes I made.  I'll be able to do that a little later or tomorrow, will update you.

**Comment by karolsojko** [comment]
Thx for helping to track this. Keep us posted as we'd like to make the experience as smooth as possible :)

**Comment by BackedUpBooty** [comment]
Hey there,

So I completely started from scratch with a newly cloned repo in a new folder, followed steps. I no longer have a pool error which is great, and after still getting `db:3306 is unavailable yet - waiting for it to start` and `auth:3000 is unavailable yet - waiting for it to start` I took a look at my firewall and opened those ports. Stuff started to happen, in the end I opened ports 3000 (for auth:3000) 3333 (exposed port in my .env), 3306 (db) and 6379 (cache).  

A bunch of stuff ran through the open log, with a few errors from the `auth` container ([see logs here](https://pastebin.com/HyDxqLDJ)) and I now have `api-gateway` and `auth-worker` giving an error of `nc: bad address 'auth'` and `auth:3000 is unavailable yet - waiting for it to start`. I haven't been able to trace that down. Far as I know I've only changed SECRET(s)/EXPOSED_PORT/PSEUDO_KEY_PARAMS_KEY/ENCRYPTION_SERVER_KEY in the envs, but just in case something else isn't correct I'm copying them below (redacted sensitive info):

**.env**
``` LOG_LEVEL=info
NODE_ENV=development

AUTH_JWT_SECRET=[PASSWD 1]

EXPOSED_PORT=3333

DB_HOST=db
DB_REPLICA_HOST=db
DB_PORT=3306
DB_USERNAME=std_notes_user
DB_PASSWORD=[PASSWD 2]
DB_DATABASE=standard_notes_db
DB_DEBUG_LEVEL=all # "all" | "query" | "schema" | "error" | "warn" | "info" | "log" | "migration"
DB_MIGRATIONS_PATH=dist/migrations/*.js

REDIS_URL=redis://cache

SNS_TOPIC_ARN=
SNS_AWS_REGION=
SQS_QUEUE_URL=
SQS_AWS_REGION=
S3_AWS_REGION=
S3_BACKUP_BUCKET_NAME=

REDIS_EVENTS_CHANNEL=events

AUTH_SERVER_URL=http://auth:3000

EMAIL_ATTACHMENT_MAX_BYTE_SIZE=10485760

REVISIONS_FREQUENCY=300

# (Optional) New Relic Setup
NEW_RELIC_ENABLED=false
NEW_RELIC_APP_NAME="Syncing Server JS"
NEW_RELIC_LICENSE_KEY=
NEW_RELIC_NO_CONFIG_FILE=true
NEW_RELIC_DISTRIBUTED_TRACING_ENABLED=false
NEW_RELIC_LOG_ENABLED=false
NEW_RELIC_LOG_LEVEL=info
```
**auth.env**
```LOG_LEVEL="info"
NODE_ENV="production"

AUTH_JWT_TTL=60000
JWT_SECRET=[PASSWD 1]
LEGACY_JWT_SECRET=[PASSWD 3]

NEW_RELIC_ENABLED=false
NEW_RELIC_APP_NAME=Auth
NEW_RELIC_NO_CONFIG_FILE=true

REDIS_EVENTS_CHANNEL="auth-events"

DISABLE_USER_REGISTRATION=false

PSEUDO_KEY_PARAMS_KEY=[PASSWD 4]

ACCESS_TOKEN_AGE=5184000
REFRESH_TOKEN_AGE=31556926

MAX_LOGIN_ATTEMPTS=6
FAILED_LOGIN_LOCKOUT=3600

EPHEMERAL_SESSION_AGE=259200

# Must be a hex string exactly 32 bytes long
# e.g. feffe9928665731c6d6a8f9467308308feffe9928665731c6d6a8f9467308308
ENCRYPTION_SERVER_KEY=[32 CHARACTER HEX STRING]
```
**api-gateway.env**
``` LOG_LEVEL="info"
NODE_ENV="production"

NEW_RELIC_ENABLED=false
NEW_RELIC_APP_NAME="API Gateway"
NEW_RELIC_NO_CONFIG_FILE=true

SYNCING_SERVER_JS_URL="http://syncing-server-js:3000"
AUTH_SERVER_URL="http://auth:3000"
```

**Comment by karolsojko** [comment]
I can see the issue is with the `alter table` migration that has not run on auth - I've fixed it in https://github.com/standardnotes/auth/commit/740cccd817aed1376d9682e400d5f29c3a3e33d6

@BackedUpBooty are you running this on a clean setup or on a pre-existing database?

If it is a clean setup and you've got nothing to lose I would recommend doing `./server.sh cleanup` and `./server.sh update`

**Comment by BackedUpBooty** [comment]
Thanks for your help in this. 

It's a brand new setup so I did a cleanup and update this is now working. Appreciate your time! Closing now as it has been resolved, I can resolve localhost:3333/healthcheck which returns a screen saying 'OK'.
